Addition of irrigation water pumping costs for India

vartika271987 opened the pull request:
## :bird: Purpose of this PR :bird:

- This PR includes the addition of irrigation water pumping costs for India, including input data as well as code changes. This also includes switches for policy shocks for India 

## :wrench: Checklist for PR creator :wrench:

- [X] Labeling pull request correctly [from the label list](https://github.com/magpiemodel/magpie/labels).

* Medium risk : New realization / Changes to existing realization / Other changes which don't modify default.cfg

- [X] Providing additional information based on PR label

* Medium risk : Default run based on the current version of the fork from which PR is made
* Cropland at Global level
![image](https://user-images.githubusercontent.com/51452385/173556943-d55e8d61-a47c-4219-8f4c-4d972d97493d.png)

*Water Use at Global level - difference mainly due to India
![image](https://user-images.githubusercontent.com/51452385/173557083-c4eda2ea-aff3-491a-b7a0-3cbde6863258.png)
![image](https://user-images.githubusercontent.com/51452385/173557383-5ba3eb9b-43a7-48e4-9d7d-169cc96a9f00.png)

*Crop Production 
![image](https://user-images.githubusercontent.com/51452385/173557495-49e9d54d-cb58-40d0-b98e-82e72b4e085b.png)

*Crop Production in India 
![image](https://user-images.githubusercontent.com/51452385/173557580-e139181f-2afe-49c8-aea0-0b456842386d.png)


- [ ] :chart_with_downwards_trend: Performance loss/gain from current default behavior :chart_with_upwards_trend:
  * Current develop branch default : 31.1 mins
  * This PR's default :  29.5 mins

- [x] Added changes to `CHANGELOG.md`
- [X] Compilation check (model starts without compilation errors - use `gams main.gms action=c` in model folder for testing).
- [X] No hard coded numbers and cluster/country/region names.
- [X] The new code doesn't contain declared but unused parameters or variables.
- [X] Where relevant, In-code comments added including documentation comments.
- [X] Made sure that documentation created with [`goxygen`](https://github.com/pik-piam/goxygen) is okay (use `goxygen::goxygen()` for testing).
- [X] Changes to [`magpie4`](https://github.com/pik-piam/magpie4) R library for post processing of model output (ideally backward compatible).
- [X] Self-review of my own code.
- [ ] For high risk runs: validation of major model indicators - Land-use, emissions, food prices, Tau.  %Delete this line in case it is not a high risk run%

## :warning: Additional notes or warnings :warning:
NA

## :rotating_light: Checklist for RSE reviewer :rotating_light:

- [ ] PR is labeled correctly.
- [ ] `CHANGELOG` is updated correctly
- [ ] No hard coded numbers and cluster/country/region names.
- [ ] No unnecessary increase in module interfaces
- [ ] All required updates in interfaces (if any) have been properly adressed in the module contracts
- [ ] In-code comments and documentation comments are satisfactory.
- [ ] model behavior/performance is satisfactory.
- [ ] Requested changes (if any) were applied correctly

## :rotating_light: Checklist for MAgPIE reviewer :rotating_light:

- [ ] PR is labeled correctly.
- [ ] `CHANGELOG` is updated correctly
- [ ] No hard coded numbers and cluster/country/region names.
- [ ] Changes to the model are scientifically sound
- [ ] In-code comments and documentation comments are satisfactory.
- [ ] model behavior/performance is satisfactory.
- [ ] Requested changes (if any) were applied correctly

mishkos reviewed (changes requested):
Please review my comments. In principle the implementation of pumping costs is straightforward, but needs better documentation. You should also explain in the module that this part is added. (and also that this is only affected India at the moment)

vartika271987 commented on config/default.cfg:
```diff
@@ -958,6 +966,12 @@ cfg$gms$s42_env_flow_fraction <- 0.2           # def = 0.2
 cfg$gms$s42_env_flow_base_fraction <- 0.05      # def = 0.05
 
+# * Year in which pumping costs shocks will be implemented
+cfg$gms$s42_shockyear <- 1995      # def = 1995
```
Done

vartika271987 commented on main.gms:
```diff
@@ -147,24 +147,24 @@ $title magpie
 *##################### R SECTION START (VERSION INFO) ##########################
```
You mean I remove the main.gms file altogether?

tscheypidi replied:
mishko means not to remove the file itself but to revert all changes you did to this file

vartika271987 commented on modules/42_water_demand/agr_sector_aug13/not_used.txt:
```diff
@@ -0,0 +1,9 @@
+# |  Contact: magpie@pik-potsdam.de
+
+name,type,reason
+vm_water_cost, variable, calculates cost of water pumping to be used in the costs module directly 
```
This declaration is made in the `not_used.txt` file in the `agr_sector_aug13` realization of the same module, as it is not the default realization currently and I have not made the same adjustments in this realization.

tscheypidi replied:
the variable `vm_water_cost` is declared in the water demand module. As it is an interface variable it has to be declared and set in every realization of that module. Currently, the other realizations won't work anymore due to a domain violation error (the costs module tries to access a variable which does not exist).
So, if you have a water demand realization which should not introduce any water costs, you still have to do two things:

- declare  the variable
- fix the variable to 0

vartika271987 commented on modules/42_water_demand/all_sectors_aug13/declarations.gms:
```diff
@@ -15,21 +15,26 @@ parameters
 * country-specific scenario switch
  p42_country_dummy(iso)             Dummy parameter indicating whether country is affected by EFP (1)
  p42_EFP_region_shr(t_all,i)        Weighted share of region with regards to EFP (1)
+ ic42_pumping_cost(i)          Parameter to capture values for pumping costs in a particular time step (USD per m^3 per yr)
```
It is unit cost. Fixed it now.

vartika271987 commented on modules/42_water_demand/all_sectors_aug13/declarations.gms:
```diff
@@ -15,21 +15,26 @@ parameters
 
 equations
  q42_water_demand(wat_dem,j)         Water withdrawals of different sectors (mio. m^3 per yr)
+ q42_water_cost(i)                Total cost of pumping irrigation water (USD)
```
Fixed it now.

vartika271987 commented on modules/42_water_demand/all_sectors_aug13/input.gms:
```diff
@@ -36,6 +36,8 @@ s42_env_flow_base_fraction         Fraction of available water that is reserved
 *                                                                    (determined in the file
 *                                                                   EFR_protection_policy.csv)
 s42_env_flow_fraction              Fraction of available water that is reserved for under protection policies (1) / 0.2 /
+s42_shockyear                       Year from which pumping costs will be implemented (1) / 2020 /
```
Done

FelicitasBeier commented on modules/43_water_availability/total_water_aug13/input.gms:
```diff
@@ -18,6 +20,3 @@ $include "./modules/43_water_availability/input/lpj_watavail_grper.cs2"
 $offdelim
 /
 ;
-$if "%c43_watavail_scenario%" == "nocc" f43_wat_avail(t_all,j) = f43_wat_avail("y1995",j);
```
they have been moved to preloop

FelicitasBeier commented on modules/42_water_demand/all_sectors_aug13/input.gms:
```diff
@@ -115,3 +117,13 @@ $ondelim
 $offdelim
 ;
+
+*Costs of pumping are taken from Cornish et.al. 2014 which are average global costs in USD
```
I don't recall that they report pumping costs directly. I thought it was a case study had different case studies looking at irrigation water prices per cubic meter that depending on the regions covered quite different aspects, and that the range between different countries is huge. 
Please clarify which prices you took precisely (and if necessary: how you transformed to pumping costs as reported in f42_pumping_cost) rather than calling it pumping costs, that I couldn't find in the publication.

vartika271987 replied:
Thanks for pointing out and I agree this was confusing. I clarify in detail in your comments below, but we are not using that publication as a reference any longer. So, I have changed the description here.

FelicitasBeier reviewed (dismissed):
I have a few doubts on the introduction of "pumping costs" only for India in the develop:

1. (content-point:) The publication does not talk about pumping costs, but only about "irrigation water prices" and these are highly dependent on very local circumstances (topology; water source; policies). They only seem to look at selected case studies and I doubt that they would be representative for "pumping costs" in India. 
2. (content-point:) Also you say that you take the "global average costs" at some point, but then only apply them to India. I think this would lead to a distortion in the model. I don't recall that "irrigation costs" in India are exceptionally high in India compared to other countries (e.g. USA, Australia, India have higher costs, right?, but would in the model then have a comparative advantage, because they don't see cost on irrigation). Given agricultural subsidies in India, one could even argue that the costs are rather lower in India compared to other locations.
3. (technical point:) why is it only introduced in the all_sectors_aug13 realization, but not in the agr_sector realization?
4. (technical point:) given that we know that we tend to underestimate the yield difference between irrigated and rainfed yields in India (because of unaccounted factors such as multiple cropping), doesn't the introduction of additional costs go into the wrong direction? 
5. (technical point:) is there a way of having 0 pumping costs (and is this still the default in the end)? (sorry, might have overlooked that in the PR);
6. (technical point:) the file is defined over t_all and i; does it only contain values for India? Which values exactly?

vartika271987 commented on modules/42_water_demand/all_sectors_aug13/presolve.gms:
```diff
@@ -46,6 +46,9 @@ if((s42_env_flow_scenario=0),
  i42_env_flows(t,j) = 0;
 Elseif(s42_env_flow_scenario=1),
+  if(m_year(t) > s42_shockyear,
+  i42_env_flows(t,j) = s42_env_flow_fraction * sum(wat_src, im_wat_avail(t,wat_src,j));
```
This was an oversight. Removed now.

vartika271987 commented on modules/43_water_availability/total_water_aug13/declarations.gms:
```diff
@@ -7,9 +7,11 @@
 
 parameters
  im_wat_avail(t,wat_src,j)        Water availability (mio. m^3 per yr)
+ p43_exo_wat_req(t_all,j)         Water demand at the time of initialization (mio. m^3 per yr)
```
Mistake. Removed.

vartika271987 commented on config/default.cfg:
```diff
@@ -35,6 +35,14 @@ cfg$input <- c(regional    = "rev4.72_h12_magpie.tgz",
 
 
+#Input files for India specific analysis
+# cfg$input <- c(cellular = "rev4.72_0306_indiaYields_05_h12_fd712c0b_cellularmagpie_c200_MRI-ESM2-0-ssp370_lpjml-8e6c5eb1.tgz",
```
We want to keep it here as these will be input files for any India specific analysis, to be used by our remaining team members. Putting it there in the default.cfg ensures it is always there and easier for team members to use in their start scripts.

vartika271987 commented on config/default.cfg:
```diff
@@ -958,6 +966,12 @@ cfg$gms$s42_env_flow_fraction <- 0.2           # def = 0.2
+# * Year in which pumping costs shocks will be implemented
+cfg$gms$s42_shockyear <- 1995      # def = 1995
+
+# * Multiplier value for different pumping cost scenarios
```
Thanks. I have now clarified it to read the following `Scalar to change pumping costs for sensitivity analysis, takes numeric values`

vartika271987 commented:
Thanks Feli. Your concern is mostly around the use of the Cornish et.al. publication for irrigation water costs, but to clarify, we are not using that reference any longer. And therefore, in the input file, we do not have any pumping costs for any of the regions except India. For India, I have calculated the pumping costs by applying conversion factors to convert energy prices of Kilowatts per hour to prices per cubic meter and for the year 2005, these costs turn out to be `0.05 cents`. We now apply this value as `pumping costs` for India in the input file as a default value as they have been obtained from existing observed data. I believe this addresses points 1 and 2. 
Point 3: Fair point. I introduced this only in `all_sectors` because that was the current default and at some point it was discussed that it was the more advanced and useful realization. If we still feel the need to update `agr_sectors` realization, I could add it there too. 

Point 4: Since we underestimate irrigated yields for India, we have already applied a correction factor for yields in the pre-processing for India. These data with improved India yields are the input files we will be using for India and hence I have added their paths to the default.cfg file. We have tested that using the improved yields input data for India with the default pumping costs of `0.05` cents does not bring much change in the model outcomes. 

Point 5: The default pumping costs are 0 for all the regions except India where they are `0.05 cents`. See input file here. The description still refers Cornish et.al but I will change it now since we don't use it any longer.  
![image](https://user-images.githubusercontent.com/51452385/174226944-8c6ea211-f67e-4a62-a742-3ebd7da68e19.png)

Point 6: See input file above. Does anything need to be changed w.r.t `t` or `t_all` ?

vartika271987 commented on modules/42_water_demand/all_sectors_aug13/input.gms:
```diff
@@ -115,3 +117,13 @@ $ondelim
+
+*Costs of pumping are taken from Cornish et.al. 2014 which are average global costs in USD
+parameter
+f42_pumping_cost(t_all,i) Cost of pumping irrigation water (1)
```
Done

tscheypidi commented on CHANGELOG.md:
```diff
@@ -45,6 +45,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
  - **scripts** scripts/output/extra/emulator.R Remove dependency on deprecated R package "magpie"
 - **56_ghg_policy** Deactivated GHG emission policies were not accounted for in the MACCs module. This has been corrected by an extension of the interface `im_pollutant_prices`, which now has an additional dimension for emission sources `emis_source`. In this context some equations in `56_ghg_policy` have been simplified (sets: `emis_source_reg`, `emis_source_cell`). Also, GHG emissions from peat ...
+- **42_water_demand** Added new input data on pumping costs for India, equation to calculate water costs and scalars for policy shocks
+- **config.cfg** Added paths to input files for India country analysis 
```
```suggestion
```

tscheypidi commented on modules/42_water_demand/all_sectors_aug13/declarations.gms:
```diff
@@ -15,21 +15,26 @@ parameters
 positive variables
   vm_watdem(wat_dem,j)               Amount of water needed in different sectors (mio. m^3 per yr)
   v42_irrig_eff(j)                   Irrigation efficiency (1)
+  vm_water_cost(i)                   Cost of irrigation water (USD per m^3 per yr)
```
I think Misko is here referring to `per yr` which indeed does not sound correct. I think the proper unit should be `USD per m^3`.

tscheypidi commented on modules/42_water_demand/all_sectors_aug13/presolve.gms:
```diff
@@ -65,3 +65,8 @@ else
+*Pumping cost in the current time step
+if(m_year(t) > s42_pumping_startyear,
+ic42_pumping_cost(i) = f42_pumping_cost(t,i)*s42_multiplier;
+);
```
what happens for years before the start year with `ic42_pumping_costs`? The parameter should be initialized for these years as well. And do you actually use this feature that the start year can be changed? If not, I would suggest to skip it and just set the pumping costs for all years

vartika271987 replied:
The parameter is initialised from the beginning itself. It is only through `s42_pumping_startyear` that I allow it to take a value as determined by the scalar `s42_multiplier`. The input data file contains values from the year 1995 itself.

tscheypidi replied:
same here as in the other realization: `s42_pumping = 0` does not deactivate pumping costs

vartika271987 commented on modules/42_water_demand/input/files:
```diff
@@ -4,3 +4,4 @@ lpj_envflow_grper.cs2
 watdem_nonagr_grper.cs3
 f42_env_flow_policy.cs3
 f42_wat_req_fao.csv
+f42_pumping_cost.cs4
```
Yes it is now a part of the default model inputs.

vartika271987 commented on scripts/start/projects/india_water.R:
```diff
@@ -0,0 +1,233 @@
+#  cfg$gms$s42_shock_year <- 2020                #def = 1995
+#  cfg$gms$s42_shock_scalar<- 0.4                #def = 0.4
+
+#start_run(cfg, codeCheck=F)
```
Understood. I also changed the start script completely now as it had irrelevant material for my test runs. I have renamed it to `fable_india` to also contain the input data files for project requirements.

tscheypidi commented:
This is an important point Feli brought up and I somehow missed. I somehow was (apparently incorrectly) assuming that in a default setup we have globally 0 pumping costs and that only in your specific India setup you introduce costs for India. As it is crucial not to introduce systematic biases towards certain regions in a global analysis it is critically important that in the default configuration there are 0 pumping costs globally, including India.

vartika271987 commented:
Hi Feli and Jan,

I understand the concerns around implementing these costs only for India but to clarify, these values are too small to bring about any significant differences in model outcomes overall, as I have shared in the results above. Adding more results below where you will see that the results of the model do not differ much in the `default` (with 0 pumping costs for all regions, including India) and `default_pc` where the costs are included only for India. I agree that several other regions have water costs as well as to our best we should include them, but until we don't have these values (as it would take some time to obtain, calculate and integrate them), we could have this setup that will also allow me to run scenarios for my paper and our India team to include in their multiple analyses as well. I am proposing to make `default_pc` as the new `default`. 

Cropland 
![image](https://user-images.githubusercontent.com/51452385/174980579-50e628fb-2492-4b98-883a-51ec8a62a7e1.png)

Water use (the effect at Global level is only because of India and other regions have no impact) 
![image](https://user-images.githubusercontent.com/51452385/174980757-93d91eb1-9765-4e1b-ba4c-b3151423a549.png)

Crop Production (small impacts for India, negligible for other regions)
![image](https://user-images.githubusercontent.com/51452385/174981021-78ce80ba-4173-43a6-92fc-b274f207086d.png)

CH4 Emissions 
![image](https://user-images.githubusercontent.com/51452385/174980403-36bd9f64-c2a5-4114-836b-7c98c66672d4.png)

Trade (change observed only in LAM and India)
![image](https://user-images.githubusercontent.com/51452385/174982026-eb9e24df-9f99-4613-9698-23e25da2118f.png)

tscheypidi commented:
Hi Vartika,

for our standard runs it is important to avoid systematic biases where possible, independent of whether they have a significant impact on the results or not. Hence, there must be a setting used as default in which all regions are treated equally (in this particular case a scenario in which all pumping costs are 0). 
As discussed earlier it is ok to have a (non-default) scenario in which there are costs for india but not for other regions so that you can use it for your studies. What we are asking for is that 1) this scenario is not default 2) there is a default scenario with 0 pumping costs. 
From my perspective this should be rather easy to implement and should not conflict with your paper plans.

vartika271987 commented:
Thank you all for the discussion on this. I have now made the change so that default pumping costs are 0 for all the regions and added the switch option in `fable_india` start script where we allow the value of pumping costs for India to be used for India-specific analyses. You may please review again now and approve if all looks good @mishkos @tscheypidi @FelicitasBeier

mishkos commented on config/default.cfg:
```diff
@@ -916,16 +907,16 @@ cfg$gms$s40_pasture_transport_costs <- 0             # def = 0
 
+# * switch for initialization area
+# * (LUH2v2): area equipped for irrigation based on LUH2v2
+# * (Siebert): area equipped for irrigation from Siebert et al.
```
why is this moved to the end here and not before the line where this switch is used?

vartika271987 replied:
This was a mistake. My mouse playing catch. Fixed it now.

mishkos commented on modules/42_water_demand/agr_sector_aug13/declarations.gms:
```diff
@@ -15,15 +15,18 @@ parameters
 * country-specific scenario switch
  p42_country_dummy(iso)             Dummy parameter indicating whether country is affected by EFP (1)
  p42_EFP_region_shr(t_all,i)        Weighted share of region with regards to EFP (1)
+ ic42_pumping_cost(i)               Parameter to capture values for pumping costs in a particular time step (USD per m^3)
```
Here also instead of `USD` write `USD05MER`

vartika271987 replied:
Done

mishkos commented on modules/42_water_demand/agr_sector_aug13/declarations.gms:
```diff
@@ -15,15 +15,18 @@ parameters
 positive variables
   vm_watdem(wat_dem,j)               Water demand from different sectors (mio. m^3 per yr)
   v42_irrig_eff(j)                   Irrigation efficiency (1)
+  vm_water_cost(i)                   Cost of irrigation water (USD per m^3)
```
also here the unit

vartika271987 replied:
Done

mishkos commented on modules/42_water_demand/agr_sector_aug13/input.gms:
```diff
@@ -106,3 +108,13 @@ $ondelim
+
+*Costs of pumping are taken observed values in India and a conversion factor is applied to change them into USD values per cubic meter
+parameter
+f42_pumping_cost(t_all,i) Cost of pumping irrigation water (USD per m^3)
```
unit

vartika271987 replied:
Done

mishkos commented on modules/42_water_demand/agr_sector_aug13/input.gms:
```diff
@@ -106,3 +108,13 @@ $ondelim
 $offdelim
 ;
+
+*Costs of pumping are taken observed values in India and a conversion factor is applied to change them into USD values per cubic meter
```
here is a need for more documentation. Perhaps refer to your study in preparation.

vartika271987 replied:
Edited reference of forthcoming paper, god willing!

mishkos commented on modules/42_water_demand/all_sectors_aug13/declarations.gms:
```diff
@@ -15,18 +15,18 @@ parameters
  p42_country_dummy(iso)             Dummy parameter indicating whether country is affected by EFP (1)
  p42_EFP_region_shr(t_all,i)        Weighted share of region with regards to EFP (1)
- ic42_pumping_cost(i)          Parameter to capture values for pumping costs in a particular time step (USD per m^3 per yr)
+ ic42_pumping_cost(i)               Parameter to capture values for pumping costs in a particular time step (USD per m^3)
```
unit

mishkos commented on modules/42_water_demand/all_sectors_aug13/declarations.gms:
```diff
@@ -15,18 +15,18 @@ parameters
   vm_watdem(wat_dem,j)               Amount of water needed in different sectors (mio. m^3 per yr)
   v42_irrig_eff(j)                   Irrigation efficiency (1)
-  vm_water_cost(i)                   Cost of irrigation water (USD per m^3 per yr)
+  vm_water_cost(i)                   Cost of irrigation water (USD per m^3)
```
unit

vartika271987 replied:
Done

mishkos commented on modules/42_water_demand/all_sectors_aug13/input.gms:
```diff
@@ -118,9 +118,9 @@ $include "./modules/42_water_demand/input/f42_env_flow_policy.cs3"
 ;
 
-*Costs of pumping are taken from Cornish et.al. 2014 which are average global costs in USD
+*Costs of pumping are taken observed values in India and a conversion factor is applied to change them into USD values per cubic meter
```
refer to your study or add more detail on calculation.

vartika271987 replied:
Done

mishkos reviewed (changes requested):
Some minor changes requested.

tscheypidi commented on modules/42_water_demand/agr_sector_aug13/equations.gms:
```diff
@@ -7,13 +7,15 @@
  + sum(kli,vm_prod(j2,kli) * ic42_wat_req_k(j2,kli) * v42_irrig_eff(j2));
 
+ q42_water_cost(i2)  ..
+   vm_water_cost(i2) =e= sum(cell(i2,j2),vm_watdem("agriculture",j2))*ic42_pumping_cost(i2);
```
This equation shows up as recently added. How were the pumping costs considered before?

tscheypidi reviewed (changes requested):
I think we are getting close to have this PR ready for approval. Most important step before that is to have a clear interface to switch off/on pumping costs in the model.

vartika271987 commented on modules/42_water_demand/all_sectors_aug13/input.gms:
```diff
@@ -118,9 +118,9 @@ $include "./modules/42_water_demand/input/f42_env_flow_policy.cs3"
+*Costs of pumping are taken observed values in India and a conversion factor is applied to change them into USD values per cubic meter
 parameter
-f42_pumping_cost(t_all,i) Cost of pumping irrigation water (1)
+f42_pumping_cost(t_all,i) Cost of pumping irrigation water (USD per m^3)
```
Done

vartika271987 commented on config/default.cfg:
```diff
@@ -916,16 +907,16 @@ cfg$gms$s40_pasture_transport_costs <- 0             # def = 0
 
+# * switch for initialization area
+# * (LUH2v2): area equipped for irrigation based on LUH2v2
+# * (Siebert): area equipped for irrigation from Siebert et al.
```
Error. Fixed it.

vartika271987 commented on modules/42_water_demand/agr_sector_aug13/not_used.txt:
```diff
@@ -6,4 +6,3 @@
 # |  Contact: magpie@pik-potsdam.de
 
 name,type,reason
-vm_water_cost, variable, calculates cost of water pumping to be used in the costs module directly 
```
Done

vartika271987 commented on config/default.cfg:
```diff
@@ -998,10 +989,11 @@ cfg$gms$s42_env_flow_fraction <- 0.2           # def = 0.2
 
 # * Scalar to change pumping costs for sensitivity analysis, takes numeric values
-cfg$gms$s42_multiplier <- 1      # def = 1
+# * Default value is 0 for all regions, set value to 1 for India default analysis
```
Thanks for the suggestion  - it does make sense for clarity. I have added a switch now `s42_pumping` that takes 0 or 1 and is 0 in default. Only when it is set to 1 will the calculations around multiplier or multiplier start year will occur, which will also be 0 just in case someone accidentally turns the switch on. For India, we would have to set the switch to 1 AND `s42_multiplier` to 1 to process pumping costs for India.

mishkos reviewed (approved):
Looks good from my side.

tscheypidi commented on modules/42_water_demand/agr_sector_aug13/presolve.gms:
```diff
@@ -61,3 +61,15 @@ else
+  if(m_year(t) > s42_multiplier_startyear,
+  ic42_pumping_cost(i) = f42_pumping_cost(t,i)*s42_multiplier;
+  );
+);
```
if `s42_pumping` is set to 0, pumping costs will be **active** as you do set pumping costs and then use these pumping costs in the equations! You need to set `ic42_pumping_cost(i) = 0;` for the case of `s42_pumping = 0`!

tscheypidi commented on CHANGELOG.md:
```diff
@@ -47,6 +47,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
  - **35_natveg** implementation of land protection moved to new module `22_land_conservation`
  - **modules** Moved interface `vm_carbon_stock` from 52_carbon to 56_ghg_policy
  - **scripts** scripts/output/extra/emulator.R Remove dependency on deprecated R package "magpie"
+- **56_ghg_policy** Deactivated GHG emission policies were not accounted for in the MACCs module. This has been corrected by an extension of the interface `im_pollutant_prices`, which now has an additional dimension for emission sources `emis_source`. In this context some equations in `56_ghg_policy` have been simplified (sets: `emis_source_reg`, `emis_source_cell`). Also, GHG emissions from peat ...
```
This line seems to be duplicated by accident. It already exists 2 lines down below

tscheypidi reviewed (changes requested):
`s42_pumping = 0` does not work yet
